src/poker.js

```diff
@@ -1,5 +1,116 @@
+const cardPointMap = {
+  A: 14,
+  K: 13,
+  Q: 12,
+  J: 11,
+  10: 10,
+  9: 9,
+  8: 8,
+  7: 7,
+  6: 6,
+  5: 5,
+  4: 4,
+  3: 3,
+  2: 2,
+  1: 1,
+};
+
 class Poker {
-  // Write your implementation here
+  /** ----------------------------------------------------------------------------------------------------------
+   * determines the winning hand whether its a nested array of pairs or triples (tackles extension 1 + extension2)
+   * @param {array} arr takes in an array of hands (arrays with either 2 or 3 indexes (cards))
+   * @returns single array (hand)
+   */
+  winningHand(cardHandArray) {
+    let currentWinningScorePair = 0;
+    let currentWinningHandPair;
+    let currentWinningScoreTrio = 0;
+    let currentWinningHandTrio;
+
+    cardHandArray.forEach((hand) => {
+      const handIsPair = this.isPair(hand);
+      const handIsTrio = this.isTrio(hand);
+      const handCardValue = this.cardValue(hand);
+
+      if (!handIsPair) return;
+
+      if (handIsTrio && handCardValue > currentWinningScoreTrio) {
+        currentWinningScoreTrio = handCardValue;
+        currentWinningHandTrio = hand;
+      }
+
+      if (handIsPair && handCardValue > currentWinningScorePair) {
+        currentWinningScorePair = handCardValue;
+        currentWinningHandPair = hand;
+      }
+    });
+
+    if (currentWinningScorePair === 0) return [];
+
+    if (currentWinningScorePair > 0 && currentWinningScoreTrio > 0)
+      return currentWinningHandTrio;
```

Comment:
Since the lowest value 'trio' is worth more than the highest pair, you could just not care if there are any pairs here. If currentWinningScoreTrio > 0 is enough to know a trio is going to win this round.